From d4075f78f79c65231d44bf076afe5552c8436d00 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 22 May 2015 15:25:38 +0800
Subject: [PATCH] storage: add the missing rootPath while generating lvm
 initramfs

While generating lvm initramfs, we check existance of '/boot/bzImage-*'
which missing the target image's rootPath(such as '/mnt/sysimage/')

If the host image do not have '/boot/bzImage-*', it skip to generate lvm
initramfs for the target image.

Upstream-Status: Inappropriate [WRLinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/storage/__init__.py | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/pyanaconda/storage/__init__.py b/pyanaconda/storage/__init__.py
index f4410ad..1d9b064 100644
--- a/pyanaconda/storage/__init__.py
+++ b/pyanaconda/storage/__init__.py
@@ -71,8 +71,8 @@ def buildLvmInitrd(anaconda):
     if anaconda.storage.mountpoints['/'].type != 'lvmlv':
         return
 
-    for f in glob("/boot/bzImage-*"):
-        initrd = "/boot/initrd-" + f.replace("/boot/bzImage-", "")
+    for f in glob(anaconda.rootPath + "/boot/bzImage-*"):
+        initrd = "/boot/initrd-" + f.replace(anaconda.rootPath + "/boot/bzImage-", "")
         log.info("build_lvm_initramfs.sh %s" % initrd)
         iutil.execWithRedirect("build_lvm_initramfs.sh", [initrd],
                                stdout = "/dev/tty5", stderr="/dev/tty5",
-- 
1.9.1

